File: P05 - Polymorphism/Java/KucingDijual.java
package Praktikum.P05.P05b_Latihan_Polymorphism_Java_2272008_Elmosius_Suli;
// File : KucingDijual.java
// Nama : Elmosius Suli
// NRP  : 2272008
// Kelas : B
// Ket :
public class KucingDijual {
    public int CekUmur(Kucing miau){
        // semua umur 3 bulan
        return(3);
    }

    public int cekHarga(Kucing miau){
        int harga = 0;

        if(miau instanceof Harimau){
            harga = 90000;
        }else if(miau instanceof KucingBesar){
            harga = 50000;
        }else if(miau instanceof KucingKecil){
            harga = 20000;
        }else if(miau instanceof Kucing)
            harga = 15000;
        return harga;
    }
}

File: P05 - Polymorphism/Python/KucingDijual.py
from Kucing import Kucing
from Harimau import Harimau
from KucingBesar import KucingBesar
from KucingKecil import KucingKecil

class KucingDijual:
    def cek_umur(self, miau: Kucing) -> int:
        return 3
    
    def cek_jenis(self, miau: Kucing) -> str:
        jns = ""
        if isinstance(miau, Harimau):
            jns = "harimau"
        elif isinstance(miau, KucingBesar):
            jns = "kucing besar"
        elif isinstance(miau, KucingKecil):
            jns = "kucing kecil"
        elif isinstance(miau, Kucing):
            jns = "kucing"
        return jns
    
    def cek_harga(self, miau: Kucing):
        harga = 0
        
        if isinstance(miau, Harimau):
            harga = 90000
        elif isinstance(miau, KucingBesar):
            harga = 50000
        elif isinstance(miau, KucingKecil):
            harga = 20000
        elif isinstance(miau, Kucing):
            harga = 15000
        return harga
        
        